distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text.enso

```diff
@@ -45,7 +45,8 @@ type Text
     compare_to : Text -> Ordering
     compare_to self that = case that of
         _ : Text ->
-            comparison_result = Text_Utils.compare_normalized self that
+            is_normalized = self.is_normalized && that.is_normalized
+            comparison_result = Text_Utils.compare self that is_normalized
             Ordering.from_sign comparison_result
         _ -> Error.throw (Type_Error.Error Text that "that")
```

Comment:
Wouldn't it make sense to make `compare_to` a builtin instead? I don't think anything is preventing us from doing that, especially now as we are importing the ICU Normalizer in the Text implementation anyway.